hw/ip/i2c/data/i2c.hjson

```diff
@@ -493,5 +496,32 @@
         { bits: "7:0" }
       ]
     }
+    { name: "STRETCH_CTRL"
+      desc: "I2C target clock stretching control"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "0"
```

Comment:
do you think we could add bits to distinguish between read / write also?

Reply:
Do you mean, so a target can stretch the clock after each read byte (if a specified bit is asserted) or after each written byte (if another specified bit is asserted)?

Reply:
yeah that's right!  So software has the flexibility to choose which "stretching mode" to use for each direction.

Reply:
I've defined ENABLETX for stretching during read and ENABLEACQ - during write.

Reply:
wait sorry i'm a bit confused, so if software sets only `ENALBEADDR` but not the other bits, what happens?

Reply:
If software sets only ENABLEADDR, a target will stretch the clock after acquiring an address from the bus and making sure it matches its own. It won't stretch the clock after reading a data byte or writing a data byte - clock stretching in these cases is controlled by ENABLETX and ENABLEACQ, respectively. This will give software more flexibility in choosing stretching modes.